forem/forem
# Adds a Default Color Hex Value to the Admin Tags Color Field

## What type of PR is this? (check all applicable)


## Description
**The issue**:
> As a Forem admin going through and marking tags as supported, I'm going to accidentally save a tag with a black background + black text color.

**The solution**:
This PR adds default hex colors to the `color_field` on the `show.html.erb` form for `/admin/tags/:id`. The default `bg_color_hex` has been set to black, `#000000`, and the default `text_color_hex` has been set to white, `#ffffff` if a tag doesn't already have a `text_color_hex` or `bg_color_hex` set.

Additionally, I renamed the `admin_creates_new_page.rb` spec to `admin_creates_new_page_spec.rb`.

## Related Tickets & Documents
Closes https://github.com/forem/InternalProjectPlanning/issues/224

## QA Instructions, Screenshots, Recordings
To QA this PR, first ensure that you have an `admin` role, and then please do the following:

- Navigate to `/admin/tags` and click on one of the "unsupported" tags listed:
![Screen Shot 2020-10-26 at 5 23 52 PM](https://user-images.githubusercontent.com/32834804/97238883-07f34b00-17b0-11eb-902b-f2dfbb125913.png)

- Once you have navigated to `/admin/tags/:id`, scroll to the bottom of the form where you can select the "Bg color hex" and the "Text color hex" and ensure that the "Bg color hex" defaults to the color black and the "Text color hex" defaults to the color white:
![Screen Shot 2020-10-26 at 5 24 23 PM](https://user-images.githubusercontent.com/32834804/97238921-18a3c100-17b0-11eb-84ba-d2d8e4708700.png)

- While still on the `/admin/tags/:id` page, check the "Supported" box and click "Update" on the form. After clicking update, you should see that the tag has now been marked as "Supported" and has a default "Bg color hex" of black and a default "Text color hex" as white, which should be reflected in the left-hand corner: 
![Screen Shot 2020-10-26 at 5 26 26 PM](https://user-images.githubusercontent.com/32834804/97239054-628ca700-17b0-11eb-8bba-5683b9c4a6ff.png)
![Screen Shot 2020-10-26 at 5 28 05 PM](https://user-images.githubusercontent.com/32834804/97239167-9d8eda80-17b0-11eb-806b-8c5929cc2878.png)

- Navigate to the tag's page and ensure that the color displayed on the page is the same color as the tags "Bg color hex" and "Text color hex" defaults:
![Screen Shot 2020-10-26 at 5 29 33 PM](https://user-images.githubusercontent.com/32834804/97239286-d16a0000-17b0-11eb-8d86-494c5c0dd796.png)

- Try to break things! 🛠️ 

## Added tests?

      have not been included_
- [x] I need help with writing tests. I've tried a handful of different approaches, but I'm still unsure of how to test this--I was unable to write a spec that worked.

## Added to documentation?


## [optional] Are there any post-deployment tasks we need to perform?
None that I can think of!

## [optional] What gif best describes this PR or how it makes you feel?

![Pixel Paint Bucket](https://media.giphy.com/media/l2YOrCLGNcjJG63i8/giphy.gif)
